src/main/java/org/scalameter/JGen.java

```diff
@@ -0,0 +1,182 @@
+package org.scalameter;
+
+import org.scalameter.picklers.*;
+import scala.Tuple2;
+import scala.Tuple3;
+import scala.Tuple4;
+import scala.collection.immutable.List;
+import scala.collection.mutable.ArrayOps;
+import scala.runtime.AbstractFunction1;
+
+
+/** Java version of the [[org.scalameter.Gen]].
+ *  After combining and changing your generators
+ *  you should finally call the `asScala()` method.
+ *
+ *  Note that it's immutable, so every method return new JGen.
+ */
+@SuppressWarnings({"unchecked"})
+public class JGen<T> {
+  private final Gen<T> gen;
+
+  public JGen(Gen<T> gen) {
+    this.gen = gen;
+  }
+
+  public Gen<T> asScala() {
+    return this.gen;
+  }
+
+  public JGen<T> cached() {
+    return new JGen<T>(gen.cached());
+  }
+
+  public <S> JGen<Tuple2<T, S>> zip(JGen<S> that) {
+    return new JGen<Tuple2<T, S>>(this.gen.zip(that.gen));
+  }
+
+  public <S> JGen<S> map(final JGenFunction<T, S> f) {
+    return new JGen<S>(this.gen.map(new AbstractFunction1<T, S>() {
+      public S apply(T v) {
+        return f.apply(v);
+      }
+    }));
+  }
+
+
+  /* Standard factory methods. */
+
+  public static JGen<Integer> exponential(
+      String axisName, int from, int until, int factor
+  ) {
+    return new JGen<Integer>(
+        (Gen<Integer>)(Object) Gen$.MODULE$.exponential(axisName, from, until, factor)
+    );
+  }
+
+  public static JGen<Integer> range(
+      String axisName, int from, int upto, int hop
+  ) {
+    return new JGen<Integer>(
+        (Gen<Integer>)(Object) Gen$.MODULE$.range(axisName, from, upto, hop)
+    );
+  }
+
+  public static JGen<Void> none(String axisName) {
+    return new JGen<Void>((Gen<Void>)(Object) Gen$.MODULE$.unit(axisName));
+  }
+
+
+  /* Factory methods that produce cross product of multiple JGen instances. */
+
+  public static <P, Q> JGen<Tuple2<P, Q>> crossProduct(
+      JGen<P> p, JGen<Q> q
+  ) {
+    return new JGen<Tuple2<P, Q>>(
+        Gen$.MODULE$.tupled(p.gen, q.gen)
```

Comment:
Perhaps we should also rename our `tupled` generators to `crossProduct`.
I doubt there are many uses out there that we would be breaking, and it would make things consistent.